### PR TITLE
Api implementation of CDM module

### PR DESCRIPTION
**Pull request by mehmetcanay**

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -9,5 +9,5 @@ pip install fastapi
 ## Usage
 
 ```python
-uvicorn main:app --reload
+cd api && uvicorn routes:app --reload
```
Comment (tiadams) [inline backend/README.md:12]: should work with:
`uvicorn api.routes:app --reload` 

no need to change dirs

File: backend/api/routes.py [hunk 2]
```diff
@@ -1,12 +1,107 @@
...
+def swagger_redirect():
+    return RedirectResponse(url="/docs")
+
+
+@app.get("/version", tags=["info"])
+def get_current_version():
+    return app.version
+
+
+@app.get("/error", status_code=200, tags=["info"])
```
Comment (tiadams) [inline backend/api/routes.py:70]: I don't think we need / use this

File: backend/api/routes.py [hunk 3]
```diff
@@ -1,12 +1,107 @@
...
+@app.get("/cdm", tags=["info"])
+def get_cdm():
+    folder_path = "../cdm"
+    files = [file for file in os.listdir(folder_path) if file.endswith(".csv")]
+    dfs = [pd.read_csv(os.path.join(folder_path, file), keep_default_na=False) for file in files]
+    cdm = pd.concat(dfs, ignore_index=True)
+    return cdm.to_dict()
+
+
+@app.get("/cdm/available_modalities", tags=["info"])
```
Comment (tiadams) [inline backend/api/routes.py:90]: /cdm/modalities

available is implicit for GET

File: backend/api/routes.py [hunk 4]
```diff
@@ -1,12 +1,107 @@
...
+def get_available_modalities():
+    folder_path = "../cdm"
+    files = [file.replace(".csv", "") for file in os.listdir(folder_path) if file.endswith(".csv")]
+    return {idx: file for idx, file in enumerate(files)}
+
+
+@app.get("/cdm/{modality}", tags=["search"])
+def get_modality(modality: str):
+    folder_path = "../cdm"
```
Comment (tiadams) [inline backend/api/routes.py:99]: better to specify from content root

File: backend/api/visualization.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import pandas as pd
+
+
+def generate_chords(modality: str, cohorts: list[str]):
```
Comment (tiadams) [inline backend/api/visualization.py:4]: Would be good to have tests for this function, optionally great would be tests for the api endpoints as well, @cebel has some experience with that and might able to give you some pointers

File: backend/api/routes.py [hunk 6]
```diff
@@ -1,12 +1,107 @@
+import logging
+import os
+import sys
+
+sys.path.insert(0, "../../backend")
+
+from api.visualization import generate_chords
```
Comment (tiadams) [inline backend/api/routes.py:7]: Should not be in the api package imo

Comment (mehmetcanay) [reply backend/api/routes.py:7]: I can omit this but without it, for some reason, api is not recognized as a module in VSCode.

File: backend/api/visualization.py (hunk 5 above)
Comment (mehmetcanay) [reply backend/api/visualization.py]: I added a test and a testing workflow for Python 3.10 and above

File: backend/api/routes.py (hunk 6 above)
Comment (tiadams) [reply backend/api/routes.py:4]: Oh that's an issue too then but I meant the package in line 7 - I'd keep the api package for REST functions

Comment (mehmetcanay) [reply backend/api/routes.py]: Fixed it

File: backend/api/routes.py (hunk 4 above)
Comment (mehmetcanay) [reply backend/api/routes.py]: I think I did it correctly but not sure

**Review by tiadams — APPROVED** [review]
LGTM!

**Pull request merged**